source/utilities.tex

```diff
@@ -13431,6 +13460,449 @@
+\tcode{\exposid{is-callable-from}<VT>} is equal to:
+\begin{codeblock}
+is_nothrow_invocable_r_v<R, VT @\cv{}@ @\placeholder{ref}@, ArgTypes...> &&
+is_nothrow_invocable_r_v<R, VT @\placeholder{inv-quals}@, ArgTypes...>
+\end{codeblock}
+Otherwise, \tcode{\exposid{is-callable-from}<VT>} is equal to:
+\begin{codeblock}
+is_invocable_r_v<R, VT @\cv{}@ @\placeholder{ref}@, ArgTypes...> &&
+is_invocable_r_v<R, VT @\placeholder{inv-quals}@, ArgTypes...>
+\end{codeblock}
+\end{itemdescr}
+
+\indexlibraryctor{copyable_function}%
+\begin{itemdecl}
+copyable_function() noexcept;
+copyable_function(nullptr_t) noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\ensures
+\tcode{*this} has no target object.
+\end{itemdescr}
+
+\indexlibraryctor{copyable_function}%
+\begin{itemdecl}
+copyable_function(const copyable_function& f);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\ensures
+\tcode{*this} has no target object if \tcode{f} had no target object.
+Otherwise, the target object of \tcode{*this}
+is a copy of the target object of \tcode{f}.
+
+\pnum
+\throws
+Any exception thrown by the initialization of the target object.
+May throw \tcode{bad_alloc}.
+\end{itemdescr}
+
+\indexlibraryctor{copyable_function}%
+\begin{itemdecl}
+copyable_function(copyable_function&& f) noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\ensures
+The target object of \tcode{*this} is
+the target object \tcode{f} had before construction, and
+\tcode{f} is in a valid state with an unspecified value.
+\end{itemdescr}
+
+\indexlibraryctor{copyable_function}%
+\begin{itemdecl}
+template<class F> copyable_function(F&& f);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{VT} be \tcode{decay_t<F>}.
+
+\pnum
+\constraints
+\begin{itemize}
+\item
+\tcode{remove_cvref_t<F>} is not the same type as \tcode{copyable_function}, and
+\item
+\tcode{remove_cvref_t<F>} is not a specialization of \tcode{in_place_type_t}, and
+\item
+\tcode{\exposid{is-callable-from}<VT>} is \tcode{true}.
+\end{itemize}
+
+\pnum
+\mandates
+\begin{itemize}
+\item
+\tcode{is_constructible_v<VT, F>} is \tcode{true}, and
+\item
+\tcode{is_copy_constructible_v<VT>} is \tcode{true}.
+\end{itemize}
+
+\pnum
+\expects
+\tcode{VT} meets the \oldconcept{Destructible} and
+\oldconcept{CopyConstructible} requirements.
+
+\pnum
+\ensures
+\tcode{*this} has no target object if any of the following hold:
+\begin{itemize}
+\item
+\tcode{f} is a null function pointer value, or
+\item
+\tcode{f} is a null member pointer value, or
+\item
+\tcode{remove_cvref_t<F>} is a specialization of
+the \tcode{copyable_function} class template,
+and \tcode{f} has no target object.
+\end{itemize}
+Otherwise, \tcode{*this} has a target object of type \tcode{VT}
+direct-non-list-initialized with \tcode{std::forward<F>(f)}.
+
+\pnum
+\throws
+Any exception thrown by the initialization of the target object.
+May throw \tcode{bad_alloc} unless \tcode{VT} is
+a function pointer or a specialization of \tcode{reference_wrapper}.
```

Comment:
```suggestion
a pointer or a specialization of \tcode{reference_wrapper}.
```
This is what is used in the next two constructors.
This suggestion should be editorial given
the constraint of _`is-callable-from`_ and
the fact that only pointers that point to functions can be callable.

Reply:
A question for the authors, but func ptrs are special cased in the rules for rval refs, so... intended?